Userland/Libraries/LibWeb/SVG/SVGDecodedImageData.cpp

```diff
@@ -98,19 +100,26 @@ ErrorOr<NonnullRefPtr<SVGDecodedImageData>> SVGDecodedImageData::create(Page& ho
 
     MUST(document->append_child(*svg_root));
 
-    return adopt_nonnull_ref_or_enomem(new (nothrow) SVGDecodedImageData(page, move(page_client), move(document), move(svg_root)));
+    return realm.heap().allocate<SVGDecodedImageData>(realm, page, page_client, document, *svg_root);
 }
 
-SVGDecodedImageData::SVGDecodedImageData(JS::NonnullGCPtr<Page> page, JS::Handle<SVGPageClient> page_client, JS::Handle<DOM::Document> document, JS::Handle<SVG::SVGSVGElement> root_element)
-    : m_page(move(page))
-    , m_page_client(move(page_client))
-    , m_document(move(document))
-    , m_root_element(move(root_element))
+SVGDecodedImageData::SVGDecodedImageData(JS::NonnullGCPtr<Page> page, JS::NonnullGCPtr<SVGPageClient> page_client, JS::NonnullGCPtr<DOM::Document> document, JS::NonnullGCPtr<SVG::SVGSVGElement> root_element)
+    : m_page(page)
+    , m_page_client(page_client)
+    , m_document(document)
+    , m_root_element(root_element)
 {
 }
 
 SVGDecodedImageData::~SVGDecodedImageData() = default;
 
+void SVGDecodedImageData::visit_edges(Cell::Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    visitor.visit(m_page);
+    visitor.visit(m_document);
```

Comment:
Shouldn't we visit `m_page_client` and `m_root_element` too?

Reply:
yes, added missing visits.